install_deps.sh

```diff
@@ -1,15 +1,14 @@
 #!/bin/sh
+
+set -ex
+
 apt-get update
 
-apt-get install vim imagemagick optipng --yes
+apt-get install ghostscript imagemagick optipng --yes
+
+rm -rf /var/lib/apt/lists/*
 
-wget -q https://s3.amazonaws.com/sl-public-dev-assets/ghostscript-9.15.tar.gz -O /tmp/ghostscript-9.15.tar.gz
-cd /tmp
-tar -xvf /tmp/ghostscript-9.15.tar.gz
-cd /tmp/ghostscript-9.15 && ./configure && make && make install
```

Comment:
I always feel like it's better to *not* have a C compiler if you don't need one. Are there dev tools we can remove?

Reply:
The `node:x.x.x` images are based on `buildpack-deps:DEBIAN_RELEASE`, which [adds a bunch of build tools](https://github.com/docker-library/buildpack-deps/blob/51a2871c1df6a6215227c1baeba81e15f04c6cff/stretch/Dockerfile) (and git + gpg tools + curl) to the respective debian image.

Reply:
We could test the `node:x.x.x-slim` variant for the second stage which is pretty much bare debian (-man pages, -docs) plus node.js: https://github.com/nodejs/docker-node/blob/46f5203674c748b0701135c8eeea4b250b3ecb6d/10/stretch-slim/Dockerfile

I am not familiar with the node-gyp process in regards to c-implementations/bindings. And how packages with other dependencies - e.g. for sqlite - behave when the libs are missing at runtime.

Reply:
FWIW, I've not had any trouble with node-gyp and the `slim` images to date --- I think building on regular `node` and then running on `slim` is one of the main use cases for the `slim` builds.

Reply:
I'll have a look at moving to node-slim after the socket.io upgrade then.

Reply:
Let's merge this now and create a new ticket for node-slim.

Reply:
@gh2k agreed on this. I have raised https://github.com/overleaf/issues/issues/2724.